### PR TITLE
Fix AVR I2C master 1ms timeout

### PR DESCRIPTION
**Pull request by dkao**
`i2c_start()` produces a minimum `time_slice` of 1ms for use as timeout value.
The timer granularity is 1ms, it is entirely possible for `timer_count` to tick up immediately after the last timer read and falsely trigger timeout with a `>= 1` comparison.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by Kriechi** [comment]
Tested and verified on ProMicro - this fixes all glitches with the Cirque trackpads on I2C that caused sudden position jumps.

File: platforms/avr/drivers/i2c_master.c [hunk 1]
```diff
@@ -64,7 +64,7 @@ static i2c_status_t i2c_start_impl(uint8_t address, uint16_t timeout) {
 
     uint16_t timeout_timer = timer_read();
     while (!(TWCR & (1 << TWINT))) {
-        if ((timeout != I2C_TIMEOUT_INFINITE) && ((timer_read() - timeout_timer) >= timeout)) {
+        if ((timeout != I2C_TIMEOUT_INFINITE) && ((timer_read() - timeout_timer) > timeout)) {
```
Comment (KarlK90) [inline platforms/avr/drivers/i2c_master.c:67]:
```suggestion
        if ((timeout != I2C_TIMEOUT_INFINITE) && (timer_elapsed(timeout_timer) > timeout)) {
```

**Review by KarlK90 — CHANGES_REQUESTED** [review]
LGTM, as part of the correction you could swap all places to the ´timer_elapsed´ convenience function.

File: platforms/avr/drivers/i2c_master.c [hunk 2]
```diff
@@ -81,7 +81,7 @@ static i2c_status_t i2c_start_impl(uint8_t address, uint16_t timeout) {
 
     timeout_timer = timer_read();
     while (!(TWCR & (1 << TWINT))) {
-        if ((timeout != I2C_TIMEOUT_INFINITE) && ((timer_read() - timeout_timer) >= timeout)) {
+        if ((timeout != I2C_TIMEOUT_INFINITE) && ((timer_read() - timeout_timer) > timeout)) {
```
Comment (KarlK90) [inline platforms/avr/drivers/i2c_master.c:84]:
```suggestion
        if ((timeout != I2C_TIMEOUT_INFINITE) && (timer_elapsed(timeout_timer) > timeout)) {
```

File: platforms/avr/drivers/i2c_master.c [hunk 3]
```diff
@@ -132,7 +132,7 @@ int16_t i2c_read_ack(uint16_t timeout) {
 
     uint16_t timeout_timer = timer_read();
     while (!(TWCR & (1 << TWINT))) {
-        if ((timeout != I2C_TIMEOUT_INFINITE) && ((timer_read() - timeout_timer) >= timeout)) {
+        if ((timeout != I2C_TIMEOUT_INFINITE) && ((timer_read() - timeout_timer) > timeout)) {
```
Comment (KarlK90) [inline platforms/avr/drivers/i2c_master.c:135]:
```suggestion
        if ((timeout != I2C_TIMEOUT_INFINITE) && (timer_elapsed(timeout_timer) > timeout)) {
```

File: platforms/avr/drivers/i2c_master.c [hunk 4]
```diff
@@ -147,7 +147,7 @@ int16_t i2c_read_nack(uint16_t timeout) {
...
```
Comment (KarlK90) [inline platforms/avr/drivers/i2c_master.c:150]:
```suggestion
        if ((timeout != I2C_TIMEOUT_INFINITE) && (timer_elapsed(timeout_timer) > timeout)) {
```

**Review by KarlK90 — APPROVED** [review]
Thanks for the fix and quick response.

**Pull request merged**